cdap-ui/app/directives/timeline/timeline-ctrl.js

```diff
@@ -87,24 +93,64 @@ function TimelineController ($scope, LogViewerStore, LOGVIEWERSTORE_ACTIONS, myL
     this.updateStartTimeInStore(apiSettings.metric.startTime);
   };
 
+  const getStatus = () => {
+    if (!pollPromise) {
+      return;
+    }
+    if (this.runId) {
+      myLogsApi.getLogsMetadata({
+        namespace : this.namespaceId,
+        appId : this.appId,
+        programType : this.programType,
+        programId : this.programId,
+        runId : this.runId
+      }).$promise.then(
+        (res) => {
+          if (pollPromise && res.status === 'KILLED' || res.status==='COMPLETED' || res.status === 'FAILED' || res.status === 'STOPPED' || res.status === 'KILLED_BY_TIMER') {
+            dataSrc.stopPoll(pollPromise.__pollId__);
+            pollPromise = null;
+          }
+        },
+        (err) => {
+          console.log('ERROR: ', err);
```

Comment:
Not really sure what to do here. The purpose of this function is to check for the preview status, and if it's killed/stopped/completed, then stop the poll. So if we get an error while trying to get the status, maybe we can stop the poll anyways?